src/nvim/lua/treesitter.c

```diff
@@ -1043,9 +1045,16 @@ static int query_next_capture(lua_State *L)
     lua_pushinteger(L, capture.index+1);  // [index]
     push_node(L, capture.node, lua_upvalueindex(2));  // [index, node]
 
+    // Now check if we need to run the predicates
     uint32_t n_pred;
     ts_query_predicates_for_pattern(query, match.pattern_index, &n_pred);
-    if (n_pred > 0 && capture_index == 0) {
+
+    lua_getfield(L, lua_upvalueindex(4), "pattern");
```

Comment:
I think it is simpler and more consistent to check if `"active"` field is undefined

Reply:
Strangely, checking for this does not work.

Reply:
Oh right, the object is reused. But i would use a field in TSLua_cursor rather than read the Lua value back then.

Reply:
Done